src/main/java/bio/terra/service/filedata/azure/util/BlobContainerClientFactory.java

```diff
@@ -0,0 +1,188 @@
+package bio.terra.service.filedata.azure.util;
+
+import com.azure.core.credential.AzureSasCredential;
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.http.HttpClient;
+import com.azure.storage.blob.*;
+import com.azure.storage.blob.models.UserDelegationKey;
+import com.azure.storage.blob.sas.BlobSasPermission;
+import com.azure.storage.blob.sas.BlobServiceSasSignatureValues;
+import com.azure.storage.common.StorageSharedKeyCredential;
+import com.azure.storage.common.sas.SasProtocol;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.util.Locale;
+import java.util.Objects;
+import org.apache.commons.lang.StringUtils;
+
+/**
+ * A class that wraps the SDK container client and facilities the creation of SAS Urls for blobs by
+ * abstracting the different strategies that depend on the authentication mechanism.
+ */
+public class BlobContainerClientFactory {
+
+  public static final int EXPIRATION_DAYS = 7;
+  private static final HttpClient httpClient = HttpClient.createDefault();
+
+  private final BlobContainerClient blobContainerClient;
+
+  private final SasTokenGeneratorStrategy sasTokenGeneratorStrategy;
+  private final AzureSasCredential blobContainerSASTokenCreds;
+  private BlobServiceClient blobServiceClient;
+  private UserDelegationKey delegationKey;
+
+  public BlobContainerClientFactory(String accountName, String accountKey, String containerName) {
+
+    if (StringUtils.isBlank(accountName)) {
+      throw new IllegalArgumentException("The account name is either null or empty.");
+    }
+
+    if (StringUtils.isBlank(accountKey)) {
+      throw new IllegalArgumentException("The account key is either null or empty.");
+    }
+
+    if (StringUtils.isBlank(containerName)) {
+      throw new IllegalArgumentException("The container name is either null or empty.");
+    }
+
+    this.blobContainerClient =
```

Comment:
It looks like this PR uses `this.` for all field member references. This doesn't follow the existing code conventions for TDR. The TDR code only uses `this.` when a field reference would be ambiguous, which is typically in a constructor or in a setter. In this case, the field reference is in a constructor but since the name doesn't collide with any of the constructor arguments, it's not ambiguous and shouldn't have a `this.`

Reply:
Thanks. Fixed.